packages/calypso-e2e/src/lib/flows/new-post-flow.ts

```diff
@@ -35,4 +36,21 @@ export class NewPostFlow {
 		await navbarComponent.clickNewPost();
 		await GutenbergEditorPage.Expect( this.page );
 	}
+
+	/**
+	 * Starts and publishes a new post from the navbar/masterbar button.
+	 *
+	 * @returns {Promise<string>} URL to the published post.
+	 */
+	async getNewTestPost(): Promise< string > {
```

Comment:
I feel like this kind of "setup action" should not be a part of a flow model. A flow should focus on specific steps that a user takes to achieve certain result, and `getNewTestPost` feels like an entire functionality of the `NewPostFlow` model enclosed in a single method. Ideally, we should be able to setup a page to a certain state using a dedicated API which would be disconnected from the client actions, as we wouldn't want i.e. the login flow breaking the post commenting flow because the one doesn't really depend on the other to function properly. Until we have a better way of setting up the env to a certain state, I'd suggest moving these kinds of hooks to a separate place. How does `site-setup-hooks.js` sound? What do you think folks, cc @scinos @bsessions85?

Reply:
@WunderBart I moved the site setup to a flow, but the calls in this method could easily be laid out in a `before` hook preceding the test, only in that case the same steps will have to be replicated each time we want to set up a test post as part of the setup.

I agree that ideally an API call would be more suitable here, but I have not looked into how involved doing that may be. In the Selenium suite, we ran into a similar issue with the shopping cart and a few other tests, so figuring out a nicely encapsulated method to programmatically setup then teardown a test site would benefit both suites at this point in time.

To have a test post setup flow is more of a temporary workaround until we figure out the above. In the meantime, specs related to `Like` post are at risk of being flakey due to the non-deterministic behavior.

Alternatively, I could drop this contentious part of the refactor but keep the changes to `test/e2e/specs/specs-playwright/wp-likes__post-spec.js` as changes to that file are more or less self-contained.